### PR TITLE
Invertable transformations, stablized SAC and reproduced bipedal walker

### LINKED ISSUE #484: TTP: StableTransformation

### PR DESCRIPTION
**Pull request by Haichao-Zhang**

### CODE REVIEW

File: alf/utils/dist_utils.py [hunk 1]
```diff
@@ -18,10 +18,31 @@
...
+from torch.distributions import constraints
 import torch.nn as nn
 
 import alf.nest as nest
 from alf.tensor_specs import TensorSpec
+from alf.utils import common
+
+
+def get_invertable(cls):
+    """A helper function to make the transformation invertable by turning
```
Comment (hnyu) [inline alf/utils/dist_utils.py]: What exactly does this mean?

Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: Some transformations may not be able to provide accurate inversion therefore the difference between x and g_inv(g(x)) is large. This is could lead to unstable training in practice.
Documentation is updated.

Comment (hnyu) [reply alf/utils/dist_utils.py]: I mean how does the cache work? One or two sentences can be added to the documentation.

Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: Now add more explanations on how the cache work.

File: alf/utils/dist_utils.py [hunk 2]
```diff
@@ -18,10 +18,37 @@
...
 import alf.nest as nest
 from alf.tensor_specs import TensorSpec
+from alf.utils import common
+
+
+def get_invertable(cls):
+    """A helper function to turn on the cache mechanism for transformation.
+    This is useful as some transformations (say g) may not be able to provide
+    an accurate inversion therefore the difference between x and g_inv(g(x)) is
+    large. This is could lead to unstable training in practice.
```
Comment (emailweixu) [inline alf/utils/dist_utils.py]: This is could => This could

File: alf/utils/dist_utils.py [hunk 3]
```diff
@@ -18,10 +18,37 @@
...
+    For a torch transformation y=g(x), when cache_size is set to one, the latest
+    value for (x, y) is cached and will be used later for future compuations.
```
Comment (emailweixu) [inline alf/utils/dist_utils.py]: compuations => computations

File: alf/utils/dist_utils.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: Corrected

File: alf/utils/dist_utils.py (hunk 3 above)
Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: Corrected

**Review by emailweixu — APPROVED** [review]

**Pull request merged**